libs/log/lazy_test.go

```diff
@@ -0,0 +1,25 @@
+package log_test
+
+import (
+	"testing"
+
+	"github.com/cometbft/cometbft/internal/test"
+	"github.com/cometbft/cometbft/libs/log"
+)
+
+func TestLazyHash_Txs(t *testing.T) {
+	const height = 2
+	const numTxs = 5
+	txs := test.MakeNTxs(height, numTxs)
+
+	for i := 0; i < numTxs; i++ {
+		hash := txs[i].Hash()
```

Comment:
Can you change the order here?

We want to make sure that line 17 does not call `Hash()`.

Reply:
Actually, we should not call `Hash()` at all here, you can do that in the `if`.